test/src/e2e_vm_tests/test_programs/should_pass/static_analysis/cei_pattern_violation_in_asm_block_smo/test.toml

```diff
@@ -0,0 +1,3 @@
+category = "compile"
+
+# check: $()Output messages modification after external contract interaction in function or method "deposit". Consider making all output messages modifications before calling another contract
```

Comment:
"Output messages modification" doesn't seem to capture the real problem with the `smo` opcode used after an interaction. I'd like to hear what @simonr0204 @nfurfaro @adlerjohn would have to say about the exact wording.

Reply:
Thanks @mohammadfawaz! Just in case, the test case with the `__smo` intrinsic is here: https://github.com/FuelLabs/sway/blob/1d59ec97466d884dae1fb543e32d34576f68f9a6/test/src/e2e_vm_tests/test_programs/should_pass/static_analysis/cei_pattern_violation_smo_intrinsic/src/main.sw. The standard library `send_typed_message` function calls the intrinsic inside its `while`-loop.

Reply:
It seems like the main issue with `smo` is that for something like a bridge, if there's a re-entrancy vulnerability in a function / chain of functions that sends a message allowing withdrawal of tokens from fuel, multiple messages could possibly be sent when there is supposed to be only a single message, allowing withdrawal of more than a users balance of tokens.
Perhaps changing the wording of "Output messages modification" to something like:
"Output Message sent after external contract interaction ..."
"Consider sending all output messages before calling another contract."

Reply:
@nfurfaro Indeed this seems much nicer, thank you! Done